lesson_13/zoldi_razvan/lesson_13_zoldi_razvan.py

```diff
@@ -0,0 +1,72 @@
+from random import randrange
+
+
+class PerimeterMixin:
+
+    def __init__(self, *args):
+        self.sides = args
+
+    def perimeter(self):
+        # return f'the perimeter of triangle with {len(self.sides)} sides is {sum(self.sides)}'
+        if len(self.sides) == 3:
+            return f'the perimeter of triangle is {sum(self.sides)}'
+        elif len(self.sides) == 4:
+            return f'the perimeter of square is {sum(self.sides)}'
+        elif len(self.sides) == 5:
+            return f'the perimeter of pentagon is {sum(self.sides)}'
+        elif len(self.sides) == 6:
+            return f'the perimeter of hexagon is {sum(self.sides)}'
+
+
+class Polygons(PerimeterMixin):
+
+    def __init__(self, *args):
+        super().__init__()
+        self.sides = args
+
+    def __str__(self):
+        no_of_sides = len(self.sides)
+        return '{} is the number of sides'.format(no_of_sides)
+
+    def display(self):
+        for side_index, length in enumerate(self.sides, start=1):
+            print('Side {} with length: {}'.format(side_index, length))
+
+
+class Triangle(Polygons):
+    # `s(s-a)(s-b)(s-c) ** 0.5` where s = `(a+b+c) / 2`
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def area(self):
+        s1, s2, s3 = self.sides
+        s_p = sum(self.sides) / 2
+        return (s_p * (s_p - s1) * (s_p - s2) * (s_p - s3)) ** 0.5
+
+
+class Square(Polygons):
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def area(self):
+        side, *_ = self.sides
+        return side ** 2
+
+    def from_area(self):
```

Comment:
`from_area` is supposed to be a `@classmethod` that should return an instance of Square. You can do this by returning cls(s1,s2,s1,s2) or Square(s1,s2,s1,s2)

Reply:
I've changed and now is @classmethod. That method at my code is working in this way: taken an given area, generate a random side and then based on this generate the other rectangle side that multiplied with random to obtain given area
I'm not forgetting: "When is necessary to explain a code you did something wrong"

Reply:
@zoldirazvan  The logic is really good. The only issue is that you are printing some string instead of returning a new object. As I mentioned here you need to return a new object

> `from_area` is supposed to be a `@classmethod` that should return an instance of Square. You can do this by returning cls(s1,s2,s1,s2) or Square(s1,s2,s1,s2)

I am closing this PR. Please make the changes and open a new one